scripts/analysis/strong_scaling.py

```diff
@@ -0,0 +1,308 @@
+import os
+import pathlib
+import re
+import sys
+from collections import defaultdict
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from specialcouscous.utils.plot import (
+    AVERAGE_KWARGS,
+    ERROR_KWARGS,
+    GLOBAL_ERROR_KWARGS,
+    GLOBAL_LINE_KWARGS,
+    INDIVIDUAL_KWARGS,
+    LINE_KWARGS,
+)
+from specialcouscous.utils.slurm import time_to_seconds
+
+pd.set_option("display.max_rows", None)
+
+
+if __name__ == "__main__":
+    # Get the root directory where results are stored from command line.
+    root_dir = sys.argv[1]
+    data_set = root_dir.split(os.sep)[-1]
+    # Dictionary to store the values grouped by (dataset, number_of_tasks, data seed)
+    results = defaultdict(list)
+    data_set = root_dir.split(os.sep)[-1]
```

Comment:
```suggestion
    data_set = root_dir.split(os.sep)[-1]
    # Dictionary to store the values grouped by (dataset, number_of_tasks, data seed)
    results = defaultdict(list)
```
Duplicate definition of `data_set`?